parallel/parallel_collections.py

```diff
@@ -30,9 +35,9 @@ def __init__(self, predicate):
         
     def __call__(self, item):
         if self.predicate(item):
-            return (True, item)
+            return EvalResult(value=True, arg=item)
         else:
-            return (False, None)
+            return EvalResult(value=False, arg=None)
 
         
```

Comment:
Good point! I also considered about your first proposal (value -> result, arg -> item), but not adopted it because I felt it is unclear at the point of your mentioned. And again I also thought about (value -> bool, arg -> item). In my opinion, bool it better. But I know there are people who don't like to use a name of built-in type/method/etc as a name of variable. So I decided to use (value, arg).
  If you don't mind about it, I'd like to use (value -> bool, arg -> item).